lib/tapioca/dsl/extensions/rubocop.rb

```diff
@@ -0,0 +1,46 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "rubocop"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Dsl
+    module Compilers
+      module Extensions
+        module RuboCop
+          extend T::Sig
+
+          MethodName = T.type_alias { T.any(String, Symbol) }
+
+          sig { params(name: MethodName, _args: T.untyped, defaults: T.untyped).returns(MethodName) }
+          def def_node_matcher(name, *_args, **defaults)
+            __tapioca_node_methods << name
+            __tapioca_node_methods << :"without_defaults_#{name}" unless defaults.empty?
+
+            super
+          end
+
+          sig { params(name: MethodName, _args: T.untyped, defaults: T.untyped).returns(MethodName) }
+          def def_node_search(name, *_args, **defaults)
+            __tapioca_node_methods << name
+            __tapioca_node_methods << :"without_defaults_#{name}" unless defaults.empty?
+
+            super
+          end
+
+          sig { returns(T::Array[MethodName]) }
+          def __tapioca_node_methods
+            @__tapioca_node_methods = T.let(@__tapioca_node_methods, T.nilable(T::Array[MethodName]))
+            @__tapioca_node_methods ||= []
```

Comment:
I think Sorbet supports this as a one-liner now:

```suggestion
            @__tapioca_node_methods ||= T.let([], T.nilable(T::Array[MethodName]))
```

Reply:
Great call! The feature was introduced in Sorbet v.0.5.10207 and we require only v0.5.9204.

So let's keep this construct 👍

Reply:
Perhaps worth asking: do we run CI against `v0.5.9204`? If we don't, perhaps it's worth adding that (in a separate PR) to ensure that we detect if we ever introduce a dependency on features from a newer version. Had I used the new notation, we might have unknowingly broken compatibility.

Reply:
Haha, [according to CI](https://github.com/Shopify/tapioca/actions/runs/3364568062/jobs/5579066013), that version of `sorbet-static-and-runtime` (apparently) doesn't even exist...

Reply:
This should be fine to change now, since we force a newer version of Sorbet.